src/wallet/wallet.cpp

```diff
@@ -2453,22 +2470,30 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
             if (wtx.tx->vout.size() <= outpoint.n) {
                 return false;
             }
-            // Just to calculate the marginal byte size
-            CInputCoin coin(wtx.tx, outpoint.n, wtx.GetSpendSize(outpoint.n, false));
-            nValueFromPresetInputs += coin.txout.nValue;
-            if (coin.m_input_bytes <= 0) {
-                return false; // Not solvable, can't estimate size for fee
-            }
-            coin.effective_value = coin.txout.nValue - coin_selection_params.effective_fee.GetFee(coin.m_input_bytes);
-            if (coin_selection_params.use_bnb) {
-                value_to_select -= coin.effective_value;
-            } else {
-                value_to_select -= coin.txout.nValue;
+            input_bytes = wtx.GetSpendSize(outpoint.n, false);
+            txout = wtx.tx->vout[outpoint.n];
+        }
+        if (input_bytes == -1) {
+            // The input is external. We either did not find the tx in mapWallet, or we did but couldn't compute the input size with wallet data
+            if (!coin_control.GetExternalOutput(outpoint, txout)) {
+                // Not ours, and we don't have solving data.
+                return false;
```

Comment:
This line will prevent you from adding the "Missing solving data for estimating transaction size" test that @jonatack suggests, since `SelectCoins` fails and the user will instead see "Insufficient funds".

One hacky fix may be to just return true here without selecting any coins, trusting that `CreateTransaction' will fail shortly thereafter. Proper fixes might be to have `SelectCoins` return a richer value (gross) or to use #20040 approach which moves this effective-value computation out of `SelectCoins` entirely.

Reply:
I have some ongoing work to refactor `CreateTransaction` and `SelectCoins` which will change some order here as well as the return values, so it might be better to push this (entire PR?) back to after that.